src/url.js

```diff
@@ -71,7 +69,7 @@ export function parseUrl(url) {
 
   const fromCache = cache[url];
   if (fromCache) {
-    return fromCache;
+    return Object.assign({}, fromCache);
```

Comment:
Lets instead copy the return value here:
https://github.com/ampproject/amphtml/blob/master/3p/integration.js#L291

On Mon, Aug 29, 2016 at 5:44 PM, Yuxuan Zhou <notifications@github.com>
wrote:

> In src/url.js
> <https://github.com/ampproject/amphtml/pull/4766#discussion_r76714281>:
>
> > @@ -71,7 +69,7 @@ export function parseUrl(url) {
> >
> >    const fromCache = cache[url];
> >    if (fromCache) {
> > -    return fromCache;
> > +    return Object.assign({}, fromCache);
>
> https://github.com/ampproject/amphtml/blob/master/3p/integration.js#L466
> Here parentLocation comes from parseUrl and is cached.
> Do you mean we want to differ from first created url object and cached url
> object?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ampproject/amphtml/pull/4766/files/20b33a1dd19e21a07c03edc6cad9f81e8d1376fc#r76714281>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAFeT9S6OnhrPxTp6RfbQUSrc1hay924ks5qk30CgaJpZM4JwBMj>
> .
>